tests on python 2.7 started to fail within the past month

not sure yet what triggered it. https://travis-ci.org/nipy/heudiconv/builds/574912662 is the first build of master to exhibit it, but changes are unrelated.

```
=================================== FAILURES ===================================
_ test_reproin_largely_smoke[--files /home/travis/build/nipy/heudiconv/heudiconv/tests/data-reproin] _
/home/travis/build/nipy/heudiconv/heudiconv/tests/test_heuristics.py:72: in test_reproin_largely_smoke
    runner(args)
/home/travis/build/nipy/heudiconv/heudiconv/cli/run.py:128: in main
    process_args(args)
/home/travis/build/nipy/heudiconv/heudiconv/cli/run.py:335: in process_args
    dcmconfig=args.dcmconfig,)
/home/travis/build/nipy/heudiconv/heudiconv/convert.py:207: in prep_conversion
    dcmconfig=dcmconfig,)
/home/travis/build/nipy/heudiconv/heudiconv/convert.py:290: in convert
    bids_options, tmpdir, dcmconfig)
/home/travis/build/nipy/heudiconv/heudiconv/convert.py:439: in nipype_convert
    eg = convertnode.run()
/home/travis/build/nipy/heudiconv/venv/lib/python2.7/site-packages/nipype/pipeline/engine/nodes.py:473: in run
    result = self._run_interface(execute=True)
/home/travis/build/nipy/heudiconv/venv/lib/python2.7/site-packages/nipype/pipeline/engine/nodes.py:564: in _run_interface
    return self._run_command(execute)
/home/travis/build/nipy/heudiconv/venv/lib/python2.7/site-packages/nipype/pipeline/engine/nodes.py:671: in _run_command
    rebase=str2bool(self.config['execution']['use_relative_paths']))
/home/travis/build/nipy/heudiconv/venv/lib/python2.7/site-packages/nipype/pipeline/engine/utils.py:269: in save_resultfile
    setattr(result.outputs, key, val)
/home/travis/build/nipy/heudiconv/venv/lib/python2.7/site-packages/nipype/interfaces/base/traits_extension.py:400: in validate
    value = super(MultiObject, self).validate(objekt, name, newvalue)
/home/travis/build/nipy/heudiconv/venv/lib/python2.7/site-packages/traits/trait_types.py:2573: in validate
    return TraitListObject(self, object, name, value)
/home/travis/build/nipy/heudiconv/venv/lib/python2.7/site-packages/traits/trait_handlers.py:2474: in __init__
    raise excp
E   TraitError: Each element of the 'converted_files' trait of a Dcm2niixOutputSpec instance must be a pathlike object or string representing an existing file, but a value of '/tmp/dcm2niixRTkim9/convert/tmp/pytest-of-travis/pytest-0/test_reproin_largely_smoke___f0/Halchenko/Yarik/950_bids_test4/sub-phantom1sid1/ses-localizer/fmap/sub-phantom1sid1_ses-localizer_acq-3mm_phasediff_e2_ph.nii.gz' <class 'future.types.newstr.newstr'> was specified.
```

--- a/heudiconv/info.py
+++ b/heudiconv/info.py
@@ -11,7 +11,7 @@ tarballs into collections of NIfTI files following pre-defined heuristic(s)."""
 REQUIRES = [
     'nibabel',
     'pydicom',
-    'nipype>=1.0.0',
+    'nipype @ git+https://github.com/nipy/nipype.git@23379ebd997b0cca3a1feb2c2d7085d4ee9cf717',
     'pathlib',
     'dcmstack>=0.7',
 ]

--- a/heudiconv/info.py
+++ b/heudiconv/info.py
@@ -11,7 +11,7 @@ tarballs into collections of NIfTI files following pre-defined heuristic(s)."""
 REQUIRES = [
     'nibabel',
     'pydicom',
-    'nipype @ git+https://github.com/nipy/nipype.git@23379ebd997b0cca3a1feb2c2d7085d4ee9cf717',
+    'nipype @ git+https://github.com/effigies/nipype.git@0196beac33d0c8c992bcc07b83c0b368810be02d',
     'pathlib',
     'dcmstack>=0.7',
 ]

--- a/heudiconv/info.py
+++ b/heudiconv/info.py
@@ -11,7 +11,8 @@ tarballs into collections of NIfTI files following pre-defined heuristic(s)."""
 REQUIRES = [
     'nibabel',
     'pydicom',
-    'nipype @ git+https://github.com/effigies/nipype.git@0196beac33d0c8c992bcc07b83c0b368810be02d',
+    'nipype >=1.0.0; python_version > "3.0"',
+    'nipype >=1.0.0,!=1.2.1,!=1.2.2; python_version == "2.7"',
     'pathlib',
     'dcmstack>=0.7',
 ]

